pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/creator/impl/ColumnarSegmentBuildingTest.java

```diff
@@ -0,0 +1,500 @@
+  private static final String NEW_INT_COL = "newIntCol";
+
+  private File _tempDir;
+  private Schema _originalSchema;
+  private Schema _extendedSchema; // Schema with additional columns
+  private TableConfig _tableConfig;
+  private List<GenericRow> _testData;
+
+  @BeforeClass
+  public void setUp() throws IOException {
+    _tempDir = new File(TEMP_DIR, "ColumnarSegmentBuildingTest");
+    FileUtils.deleteQuietly(_tempDir);
+    _tempDir.mkdirs();
+
+    // Create original schema
+    _originalSchema = new Schema.SchemaBuilder()
+            .addSingleValueDimension(STRING_COL_1, FieldSpec.DataType.STRING)
+            .addSingleValueDimension(STRING_COL_2, FieldSpec.DataType.STRING)
+            .addSingleValueDimension(INT_COL_1, FieldSpec.DataType.INT)
+            .addSingleValueDimension(INT_COL_2, FieldSpec.DataType.INT)
+            .addSingleValueDimension(LONG_COL, FieldSpec.DataType.LONG)
+            .addSingleValueDimension(FLOAT_COL, FieldSpec.DataType.FLOAT)
+            .addSingleValueDimension(DOUBLE_COL, FieldSpec.DataType.DOUBLE)
+            .addSingleValueDimension(BIG_DECIMAL_COL, FieldSpec.DataType.BIG_DECIMAL)
+            .addSingleValueDimension(BYTES_COL, FieldSpec.DataType.BYTES)
+            .addMultiValueDimension(MV_INT_COL, FieldSpec.DataType.INT)
+            .addMultiValueDimension(MV_STRING_COL, FieldSpec.DataType.STRING)
+            .addDateTime(TIME_COL, FieldSpec.DataType.LONG, "1:MILLISECONDS:EPOCH", "1:MILLISECONDS")
+            .build();
+
+    // Create extended schema with additional columns
+    _extendedSchema = new Schema.SchemaBuilder()
+            .addSingleValueDimension(STRING_COL_1, FieldSpec.DataType.STRING)
+            .addSingleValueDimension(STRING_COL_2, FieldSpec.DataType.STRING)
+            .addSingleValueDimension(INT_COL_1, FieldSpec.DataType.INT)
+            .addSingleValueDimension(INT_COL_2, FieldSpec.DataType.INT)
+            .addSingleValueDimension(LONG_COL, FieldSpec.DataType.LONG)
+            .addSingleValueDimension(FLOAT_COL, FieldSpec.DataType.FLOAT)
+            .addSingleValueDimension(DOUBLE_COL, FieldSpec.DataType.DOUBLE)
+            .addSingleValueDimension(BIG_DECIMAL_COL, FieldSpec.DataType.BIG_DECIMAL)
+            .addSingleValueDimension(BYTES_COL, FieldSpec.DataType.BYTES)
+            .addMultiValueDimension(MV_INT_COL, FieldSpec.DataType.INT)
+            .addMultiValueDimension(MV_STRING_COL, FieldSpec.DataType.STRING)
+            .addDateTime(TIME_COL, FieldSpec.DataType.LONG, "1:MILLISECONDS:EPOCH", "1:MILLISECONDS")
+            .addSingleValueDimension(NEW_STRING_COL, FieldSpec.DataType.STRING)
+            .addSingleValueDimension(NEW_INT_COL, FieldSpec.DataType.INT)
+            .build();
+
+    // Create table config
+    _tableConfig = new TableConfigBuilder(TableType.OFFLINE)
+            .setTableName(TABLE_NAME)
+            .setTimeColumnName(TIME_COL)
+            .setInvertedIndexColumns(Lists.newArrayList(STRING_COL_1, INT_COL_1))
```

Comment:
can we also add a few more indexes on some columns, mark some columns as no dictionary to increase comprehensiveness of test